distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -2112,6 +2144,182 @@ public void testProductDownloadWithTwoUsers() throws Exception {
                 "COMPLETE");
     }
 
+    @Test
+    public void testTwoUsersSameProductRetrySuccess() throws Exception {
+
+        String filename = "product2.txt";
+
+        CometDClient cometDClient1 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "localhost", "localhost");
+        CometDClient cometDClient2 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "admin", "admin");
+
+        setupCswServerForSuccess(cometDClient1, filename);
+        setupCswServerForSuccess(cometDClient2, filename);
+
+        // Verify that we get 3 notifications: 2 retrying and 1 complete.
+        // Verify that we get 9 activity messages: started, downloading, retrying, and complete
```

Comment:
@garrettfreibott yeah I think that was the type of explanation we were looking for. It was difficult to see why those specific numbers were being used.